src/app/chat/[uuid]/page.tsx

```diff
@@ -1,23 +1,28 @@
 "use client";
 
 import { notFound, useParams } from "next/navigation";
+import { useEffect, useState } from "react";
 
+import { useChats } from "@/hooks/use-chats";
 import type { Chat } from "@/types/chat";
 
 export default function ConversationPage() {
   const parameters = useParams<{ uuid: string }>();
+  const { getChat } = useChats();
+  const [chat, setChat] = useState<Chat | null | undefined>(null);
 
-  // Get chats from local storage
-  const chats =
-    localStorage.getItem("chats") == null
-      ? []
-      : (JSON.parse(localStorage.getItem("chats")!) as Chat[]);
+  useEffect(() => {
+    const foundChat = getChat(parameters.uuid);
```

Comment:
**suggestion**: I think calling it "targetChat" or something like that would be better, since `getChat()` returns a local storage response. If a chat with that uuid *is found* within local storage then we get the chat object, but of course a chat with that uuid may not exist, returning *null* in that case - therefore that chat *will not* be always found